HW4/task1.py

```diff
@@ -0,0 +1,32 @@
+# Создайте несколько переменных заканчивающихся и не оканчивающихся на «s».
+# Напишите функцию, которая при запуске заменяет содержимое переменных оканчивающихся на s
+# (кроме переменной из одной буквы s) на None.
+# Значения не удаляются, а помещаются в одноимённые переменные без s на конце.
+
+def remove_s():
+    glob_dict = globals()
+    for var in list(glob_dict):
```

Comment:
Итерирование словаря и так осуществляется по ключам.
```suggestion
    for var in glob_dict:
```

Reply:
> Итерирование словаря и так осуществляется по ключам.

Я пробовал именно так и поступать сначала, но тогда выскакивает исключение:

    _for var in glob_dict:
RuntimeError: dictionary changed size during iteration_

Если я правильно понял, это связано с тем, что в цикле мы добавляем новые переменные в пространство имён, что влияет на размер словаря, по ключам которого мы уже начали проходиться циклом. Именно поэтому пришлось "заковать" ключи в список.

Reply:
Теперь понятно, вы сделали все правильно. Просто можно было использовать тогда метод `keys`, либо `items`.